reviewing/license-scanner/src/getPkgInfo.ts

```diff
@@ -416,3 +416,32 @@ export async function getPkgInfo (
 
   return packageInfo
 }
+
+export interface ResolveModulePathOpts {
+  projectDir: string
+  virtualStoreDir: string
+  virtualStoreDirMaxLength: number
+  modulesDir: string
+  manifestName: string
+}
+
+export async function resolveModulePath (pkg: PackageInfo, opts: ResolveModulePathOpts): Promise<string> {
+  const pnpmConfig = await memReadLocalConfig(opts.projectDir)
+
+  if (pnpmConfig['nodeLinker'] === 'hoisted') {
+    // When node-linker=hoisted is set in the .npmrc the node_modules are set out as a flat list
+    return path.join(opts.projectDir, opts.modulesDir, opts.manifestName)
+  }
```

Comment:
This won't give the right location for all cases. Even with hoisted node_modules some dependencies might get nested. You need to get the path from the "hoistedLocations" field in `node_modules/.modules.yaml`.